src/profile/external-identities/OAuth2Providers.ts

```diff
@@ -0,0 +1,47 @@
+import { getConfig } from 'src/libs/config';
+
+export type OAuth2ProviderKey = 'github' | 'ras';
+
+export type OAuth2Provider = {
+  key: OAuth2ProviderKey;
+  name: string;
+  queryParams: {
+    scopes: string[];
+    redirectUri: string;
+  };
+  supportsAccessToken: boolean;
+  supportsIdToken: boolean;
+};
+
+export const oauth2Provider = (providerKey: OAuth2ProviderKey): OAuth2Provider => {
+  switch (providerKey) {
+    case 'github':
+      return {
+        key: providerKey,
+        name: 'GitHub',
+        queryParams: {
+          scopes: ['read:user', 'user:email'],
```

Comment:
I thought we wouldn't have to request scopes with Github apps because our dev app has [permissions](https://github.com/organizations/terra-bio/settings/apps/terra-app-dev/permissions) to grant read-only access to repo contents and metadata. We can confirm this when @Shakespeared adds the get auth token endpoint!

These are the docs I was looking at: https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/differences-between-github-apps-and-oauth-apps#what-can-github-apps-and-oauth-apps-access

Reply:
Also is the intention here that ECM would eventually use the scopes passed in as query params instead of the ones stored in its provider config?